juju/entrypoint.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+
+# use modules from the lib directory - works under an assumption that
+# the current directory is the top-level charm directory ($JUJU_CHARM_DIR)
+sys.path.append('lib')
+
+from juju.framework import Framework # noqa
+from charm import Charm # noqa
+
+
+# reuse the same db file as used by charm-helpers
+DB_PATH = os.path.join(
+    os.environ.get('CHARM_DIR', ''), '.unit-state.db')
+
+try:
+    from charmhelpers.core.hookenv import log
+except (ImportError, ModuleNotFoundError):
+    log = print
+
+
+def emit_charm_event(charm, event_name):
+    """Emits a charm event based on a Juju event name
+
+    :param charm: A charm instance to emit an event from
+    :type charm: py:class:`juju.charm.Charm`
+    :param str event_name: A Juju event name to emit on a charm
```

Comment:
This is something we should debate on the next call and get to some consensus: in general I'm a fan of code that is readable on the terminal, even if that makes it less nice to handle on, say, a web browser. The reason is because we'll be reading this on a terminal the vast majority of times, so the price of having all the formatting isn't worth it. As a concrete example, compare the above with:
```
charm -- A Charm instance to emit an event from.
event_name -- Name of even to emit.
```
Which of these would you rather read? Again, we can discuss this live in our next call.

Reply:
I would rather know the type as my experience in debugging complex code-bases with dynamic typing (such as OpenStack) is that knowing the intended type of a parameter saves a lot of time while you search.